examples/MQ/pixelSimSplit/run/scripts/test-splitMC.sh.in

```diff
@@ -56,28 +56,32 @@ until [ $LOOPING -eq 0 ]; do
     fairmq-dds-command-ui -s $DDS_ID -c c &> dds_check.log
     NOF_RUNNG=$(grep RUNNING dds_check.log | wc -l)
     NOF_READY=$(grep READY   dds_check.log | wc -l)
+    NOF_ERROR=$(grep ERROR   dds_check.log | wc -l)
     NOF_AGENT=$(dds-info -s $DDS_ID -n | grep -m 1 "$NOFNEEDEDAGENTS")
     echo "GOT $NOF_RUNNG RUNNING, $NOF_READY READY ($NOF_AGENT AGENTS)"
     if [ $NOF_RUNNG -eq $((NOFNEEDEDAGENTS - 1)) ] && [ $NOF_READY -eq 1 ]
     then
 	LOOPING=0;
     fi
-    : ;
+    if [ $NOF_EROOR -ne 0 ]
+    then
+	LOOPING=0;
+    fi
 done
 
 sleep 2
 
 echo "check dds states again"
-fairmq-dds-command-ui -s $DDS_ID -c c
+fairmq-dds-command-ui --session $DDS_ID -c c
```

Comment:
You can export `DDS_SESSION_ID` - then both DDS commands and fairmq-dds-command-ui will honor it.
See https://github.com/FairRootGroup/FairMQ/blob/master/examples/dds/fairmq-start-ex-dds.sh.in.